### ISSUE TITLE
Windows 10 dark window frame mode isn't aligned with Windows 10 native contrast ratio

### ISSUE THREAD

**Issue by Lootyhoof** [MoonchildProductions/Pale-Moon#1228]
While looking at this stuff it's appeared to me that the `darkwindowframe` code doesn't appear to use the same contrast ratio as Windows 10 itself.

Using Windows 10 build 15063 ("Creator's Update"), with a titlebar such as the below in Notepad:

![Notepad](http://i.imgur.com/XuCop9c.png)

This is displayed as the following in Pale Moon, as the `darkwindowframe` code is set to `true`:

![Pale Moon](http://i.imgur.com/fzGjo8A.png)

This appears to be the case for several of the pre-set colours available within Windows 10, and only seems to switch to the darker window title and caption buttons when on a very bright titlebar colour.

**Comment by wolfbeast** [comment]
This used to be pretty closely aligned (I tested with many of the preset colors in older versions of Win10) and it's actually using the accessibility formula to calculate. The "flip point" was mostly trial and error.

**Comment by wolfbeast** [comment]
I calculated the flip point used in the code I just added (luminance 110) which is a contrast ration according to the WCAG formula of 9.62 -- that's almost exactly the 9.5 I've put in the darkwindowframe calculation... 

I'm not privy to what formula Win10 uses.

**Comment by wolfbeast** [comment]
I poked [Technet](https://social.technet.microsoft.com/Forums/en-US/d942f9bd-7647-468f-869b-86f966bd42d0/how-is-the-contrast-ratio-calculated-for-deciding-text-should-be-black-or-white?forum=win10itprogeneral) to see if there's anyone who knows how exactly Windows does this contrast calculation.

**Comment by wolfbeast** [comment]
The creator's update won't run in my test environment, I found out (VMWare 11 doesn't like it and goes black). I don't know if they changed this from the anniversary update or not, but i can see about making another more concerted trial run to get the contrast point more fine tunes than 9.5.

FTR, it's in browser.js line 906

**Comment by wolfbeast** [comment]
I think setting the point to WCAG's 7:1 contrast ratio does the trick to get it closer to Windows' own.
See commit 8440d1a7c60d90ad3147759a09229c1d905c4fe9

**Comment by wolfbeast** [comment]
OK, so, according to someone on the dev forum, it's documented as follows:

```Javascript
function accentColorUpdated(elementWithText)
{
    var uiSettings = new Windows.UI.ViewManagement.UISettings();
    Windows.UI.Color c = uiSettings.GetColorValue(UIColorType.Accent);
    var colorIsDark (5 * c.g + 2 * c.r + c.b) <= 8 * 128;
    if (colorIsDark)
    {
        elementWithText.RequestedTheme = ElementTheme.Light;
    }
    else
    {
        elementWithText.RequestedTheme = ElementTheme.Dark;
    }     
}
```
https://docs.microsoft.com/en-us/windows/uwp/style/color
